dist/index.d.ts

```diff
@@ -523,20 +575,44 @@ declare global {
 
   export interface GPURenderPipelineDescriptor
     extends GPUPipelineDescriptorBase {
-    vertexStage: GPUProgrammableStageDescriptor;
-    fragmentStage?: GPUProgrammableStageDescriptor;
+    vertex?: GPUVertexState;
```

Comment:
Should not be optional

Reply:
I meant to ask about this, actually. Since the interface contains a mix of both the deprecated and refactored fields, it doesn't seem like I can make it required currently without preventing use of the deprecated path entirely? Or does Typescript have a workaround for that?

Reply:
Oh yes, sorry, right. I think I have an idea about this.

Reply:
Two possible solutions:

```ts
type GPURenderPipelineDescriptor = GPURenderPipelineDescriptorNew | GPURenderPipelineDescriptorOld;
```

```ts
type GPURenderPipelineDescriptor = GPUPipelineDescriptorBase & ({ /* new */ } or { /* old */ });
```

I think the first one is less weird.

Reply:
Pushed a commit for this

Reply:
Thanks! Didn't know that was a possibility in Typescript. Definitely feels better than smooshing them both into the same interface.